lib/file.go

```diff
@@ -116,7 +125,13 @@ func fsNamesFromFileStorageObject(storageObject string) (containerName string, f
 
 // Generate a clean filename with correct extension from a filename input
 func FileCleanName(filename string) string {
-	return fsCleanName(filename) + ".json"
+	if cleanWithJSONExtension {
+		return fsCleanName(filename) + ".json"
+	}
+	if filename == defaultFileStorageName {
+		return fsCleanName(filename+defaultFileStorageExt)
+	}
+	return fsCleanName(filename)
 }
```

Comment:
I'd like to see a comment about why the defaultFileStorageName is treated differently.